### ISSUE TITLE
Unable to open a project in Windows via drag&drop, when the project is stored in a GPKG file on a mapped drive

### ISSUE THREAD

**Issue by cxcandid** [qgis/QGIS#46969]
### What is the bug or the crash?

We cannot open a QGIS project in Windows via drag&drop, when the project is stored in a GPKG file on a mapped drive. QGIS complains that the URI is invalid. Everything works fine if we open the project via QGIS Browser, or move the GPKG file to a local drive. "Open From... > Geopackage" works as well.

### Steps to reproduce the issue

Store a QGIS project inside a GPKG file and put that file on a mapped drive (i.e. X:).

### Versions

QGIS 3.22.3 Windows

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by jfmoyen** [comment]
Similar issue using a physical drive on the same computer (D:). 
3.22.3-Białowieża
Révision 1628765ec7
Windows 10

**Comment by gioman** [comment]
> Similar issue using a physical drive on the same computer (D:).
> 3.22.3-Białowieża
> Révision [1628765](https://github.com/qgis/QGIS/commit/1628765ec7d40ded50bbaf1cd08220bb390e8783)
> Windows 10

@jfmoyen can't confirm this, just tested on Win10 and 3.22.3

**Comment by jfmoyen** [comment]
Interesting, so there would be something wrong with my file. Let me investigate, I'll come back to you.

**Comment by jfmoyen** [comment]
[GEO001M_exp.zip](https://github.com/qgis/QGIS/files/8023857/GEO001M_exp.zip)

This, for instance, does open happily if I drag it from C:\users\me\documents but not from D:\GISData\France\BRGM\France_1_000000

No, the D-path does not contain any space or weird characters, and it is not specially long, either.

This is the actual error I get:
![image](https://user-images.githubusercontent.com/22688234/152989850-b38b61d1-5373-4550-9444-258e5a7bb8a2.png)

Please let me know if there is something else you'd like me to try.

**Comment by gioman** [comment]
> This is the actual error I get:

@jfmoyen yes, I was too quick before, D&D from d: drive and the selecting the project saved in the GPKG will result in that error.

**Comment by cxcandid** [comment]
I can confirm that it only works from C: drive.

**Comment by nicogodet** [comment]
@cxcandid Sponsoring could be a solution indeed

**Comment by cxcandid** [comment]
In QGIS 3.30.3 (Windows) D&D works from D: drive as well, but not from USB and not from any mapped network drive.
If I open a network directory via UNC path in Windows Explorer, D&D works pretty well.

**Comment by cxcandid** [comment]
In QGIS 3.36.0 project opening from GPKG via drag&drop doesn't work from D: drive anymore 🙁. C: drive is ok and UNC path is supported.

**Comment by agiudiceandrea** [comment]
Hi @elpaso, I've seen that you previously fixed with PR https://github.com/qgis/QGIS/pull/33265 the logic for handling the project in a GeoPackage file. Anyway it seems it is still not working on Windows when the GeoPackage file is in drive different from the C: drive (actually QGIS install drive letter) and such GeoPackage file is dropped onto the QGIS main window.

I've tracked down the issue to the point that in such case `QgsGeoPackageProjectStorage::readProject` is called with uri = `geopackage://D:/file_name.gpkg?projectName=project_name` (see the double slash).
Then, in `QgsGeoPackageProjectStorage::decodeUri`, the uri string doesn't match with the QRegularExpression `(^[A-Za-z]:)`, so it is checked the existence (`QFile::exists`) of url.path() (where url = `QUrl::fromEncoded( uri.toUtf8() )` = `QUrl('geopackage://d/file_name.gpkg?projectName=project_name')` ) which in this case is `/file_name.gpkg` (see the missing drive letter) and thus QFile searches the file in C: (actually in the QGIS install drive letter) and it doesn't find it.

On the contrary, when such GeoPackage file is opened via Project->Open From->GeoPackage, then `QgsGeoPackageProjectStorage::readProject` is called with uri = `geopackage:D:/file_name.gpkg?projectName=project_name` (see the missing double slash), so it is checked the existence (`QFile::exists`) of url.path() (where url = `QUrl::fromEncoded( uri.toUtf8() )` = `QUrl('geopackage:D:/file_name.gpkg?projectName=project_name')` ) which in this case is `D:/file_name.gpkg` (see the presence of the drive letter) and it correctly finds it.

Maybe you know a simple way to fix it.

**Comment by agiudiceandrea** [comment]
Also recently reported at https://gis.stackexchange.com/questions/494572/qgis-cant-open-embedded-project-from-geopackage-via-drag-and-drop-on-non-standa.
